wagtail/wagtailadmin/static_src/wagtailadmin/js/blocks/list.js

```diff
@@ -47,10 +50,47 @@
                 }
             });
 
-            /* initialize 'add' button */
-            $('#' + elementPrefix + '-add').click(function() {
+            var addMember = function() {
                 sequence.insertMemberAtEnd(listMemberTemplate);
-            });
+            }
+
+            /* initialize 'add' button */
+            addButton.click(addMember);
+
+            function checkMinMax(){
+              if (opts.max_num && sequence.getRealCount() >= opts.max_num) {
+                  addButton.addClass('disabled');
+                  addButton.unbind('click', addMember)
+              } else {
+                  addButton.removeClass('disabled');
+                  /* unbind first to ensure we aren't binding multiple times */
+                  addButton.unbind('click', addMember)
+                  addButton.click(addMember);
+              }
+
+              if (opts.min_num && sequence.getRealCount() < opts.min_num){
+                if (!$("#" +  elementPrefix + "-min-error").length) {
+                  errorText.append(
+                    "<div id=\"" + elementPrefix + "-min-error\" class=\"help-block help-critical\">" + opts.min_num + " or more items required</div>");
+                }
+              } else {
+                $("#" +  elementPrefix + "-min-error").remove();
+              }
+            }
+
+            checkMinMax(sequence);
+
+            sequence.__postInsertMember = sequence._postInsertMember;
+            sequence._postInsertMember = function(newMember){
+              sequence.__postInsertMember(newMember);
+              checkMinMax();
+            }
```

Comment:
@gasman Here I'm only replacing the _postInsertMember with my own decorated version. So I need it exposed, but I still feel the original '_' helps to make it clear that this is not a function you call yourself, but rather it is internal to the sequence. Although in this case the double underscore looks a little weird, but it's clear.

I could potentially do:
```
orginal_postInsertMember = sequence._postInsertMember;
sequence._postInsertMember = function(newMember){
  orginal_postInsertMember(newMember);
  checkMinMax();
}
```